### PR TITLE
Increase api/middleware test coverage

### PR DESCRIPTION
**Pull request by AlexAPB**
This is part of the effort to increase code coverage to >=90%.

### CODE REVIEW

**Comment by breerly** [comment]
This is sweet! Suggest making smaller logical PRs (perhaps containing only one test per PR) rather than collecting everything here - that'll give you more momentum and won't hold up progress until this big PR is complete.

File: api/middleware/inbound_test.go [hunk 1]
```diff
@@ -26,13 +26,15 @@ import (
...
 	"testing"
 
-	"github.com/golang/mock/gomock"
-	"github.com/stretchr/testify/assert"
 	"go.uber.org/yarpc/api/middleware"
 	"go.uber.org/yarpc/api/transport"
 	"go.uber.org/yarpc/api/transport/transporttest"
 	"go.uber.org/yarpc/encoding/raw"
 	"go.uber.org/yarpc/internal/testtime"
+
```
Comment (peter-edge) [inline api/middleware/inbound_test.go]: No newline

Comment (AlexAPB) [reply api/middleware/inbound_test.go]: Are we not dividing imports? I thought we followed the pattern of:
```
... go library imports ...

... project imports ...

... external imports ...
```

**Comment by AlexAPB** [comment]
@breerly I wasn't going to add any more tests to this package, but I'll break up the other ones I'm working on since I already got a stamp here.

**Comment by breerly** [comment]
Just a suggestion! If this is it then it's probably coo.

File: api/middleware/inbound_test.go (hunk 1 above)
Comment (breerly) [reply api/middleware/inbound_test.go]: @AlexAPB has the right of it here I believe :)

Comment (peter-edge) [reply api/middleware/inbound_test.go]: I thought we agreed for yarpc-go we weren't doing this anymore @abhinav

**Pull request merged**